File: CivitasRuby/pkg/CivitasRuby-0.0.1/lib/sorpresa_salir_carcel.rb
#encoding:utf-8

require_relative 'jugador'
require_relative 'tablero'
require_relative 'casilla'
require_relative 'mazo_sorpresas'
require_relative 'sorpresa'
require 'sorpresa.rb'

module Civitas
    class SorpresaSalirCarcel < Sorpresa
        def initialize(mazo, texto)
            super(texto)
            @mazo=mazo
        end

        def aplicar_a_jugador(actual, todos)
            if(jugador_correcto(actual,todos))
                informe(actual, todos)
                encontrado=false
                i=0
                while !encontrado && i < todos.length
                    if(todos[i].tiene_salvoconducto)
                        encontrado=true
                    end
                    i += 1
                end
                
                if !encontrado
                    todos[actual].obtener_salvoconducto(self)
                    salir_del_mazo
                end
            end
        end

        def salir_del_mazo
            @mazo.inhabilitar_carta_especial(self)
        end
            
        def usada
            @mazo.habilitar_carta_especial(self)
        end
        
        public_class_method :new
    end
end

File: CivitasJava/src/civitas/SorpresaSalirCarcel.java
package civitas;

import java.util.ArrayList;

public class SorpresaSalirCarcel extends Sorpresa{

    SorpresaSalirCarcel(MazoSorpresas mazo, String texto){
        super(texto);
        this.mazo=mazo;
    }

    @Override
    void aplicarAJugador(int actual, ArrayList<Jugador> todos){
        if(jugadorCorrecto(actual,todos)){
            informe(actual,todos);
            boolean encontrado=false;
            for(int i=0; i<todos.size() && !encontrado;i++){
                if(todos.get(i).tieneSalvoconducto()){
                    encontrado=true;
                }
            }
            
            if(!encontrado){
                todos.get(actual).obtenerSalvoconducto(this);
                salirDelMazo();
            }
        }
    }

    void salirDelMazo(){
        mazo.inhabilitarCartaEspecial(this);
    }

    void usada(){
        mazo.habilitarCartaEspecial(this);
    }
}